lib/core/src/display/navigation/navigator.rs

```diff
@@ -0,0 +1,144 @@
+        let velocity           = zero();
+        let position           = camera.position();
+        let kinematics         = KinematicsProperties::new(position, velocity, zero(), mass);
+        let spring_coefficient = 10000.0;
+        let fixed_point        = camera.position();
+        let spring             = SpringProperties::new(spring_coefficient, fixed_point);
+        let drag               = DragProperties::new(1500.0);
+        let properties         = PhysicsProperties::new(kinematics, spring, drag);
+        let camera             = camera.object;
+        let steps_per_second   = 60.0;
+        let properties_clone   = properties.clone();
+        let simulator          = PhysicsSimulator::new(
+            &mut event_loop,
+            steps_per_second,
+            camera,
+            properties_clone
+        );
+        (simulator, properties)
+    }
+
+    fn start_navigator_events
+    ( dom:&DOMContainer
+    , camera:Camera
+    , min_zoom:f32
+    , max_zoom:f32
+    , zoom_speed:f32
+    , mut properties:PhysicsProperties) -> Result<NavigatorEvents> {
+        let dom_clone            = dom.clone();
+        let camera_clone         = camera.clone();
+        let mut properties_clone = properties.clone();
+        let panning_callback     = move |pan: PanEvent| {
+            // base_distance is a distance where the camera covers all the UI.
+            let base_distance = dom_clone.dimensions().y / 2.0 * camera_clone.get_y_scale();
+            // We can then scale the movement based on the camera distance. If we are closer, the
+            // movement will be slower, if we are further, the movement is faster.
+            let scale         = camera_clone.position().z / base_distance;
+
+            let x = pan.movement.x * scale;
+            let y = pan.movement.y * scale;
+            let z = 0.0;
+
+            properties_clone.mod_spring(|spring| { spring.fixed_point += Vector3::new(x, y, z); });
+        };
+
+        let dom_clone = dom.clone();
+        let zoom_callback = move |zoom:ZoomEvent| {
+            if let CameraType::Perspective(persp) = camera.camera_type() {
+                let point      = zoom.focus;
+                let normalized = normalize_point2(point, dom_clone.dimensions());
+                let normalized = normalized_to_range2(normalized, -1.0, 1.0);
+
+                // Scale X and Y to compensate aspect and fov.
+                let x                     = -normalized.x * persp.aspect;
+                let y                     =  normalized.y;
+                let z                     = camera.get_y_scale();
+                let direction             = Vector3::new(x, y, z).normalize();
+                let mut position          = properties.spring().fixed_point;
+                let zoom_amount           = zoom.amount * position.z;
+                position                 += direction   * zoom_amount;
+                let min_zoom              = persp.near + 1.0 + min_zoom;
```

Comment:
What is this `1.0`? All literals should be extracted to variables and described. Looking at that equation I dont understand what it is doing and if it is really necessary.

Reply:
Now that we have `min_zoom`, it is no longer necessary.